### ISSUE TITLE
[FEAT]: "forgotten password" workflow

### ISSUE THREAD

**Issue by CharlesLgn** [revoicechat/revoicechat#21]
### Description

We need a system to reset a password in case we forgot our password

**Comment by CharlesLgn** [comment]
## Forgot password recovery methods comparison table

| Method                    | Infra needed | Security   | Pros                                                                 | Cons                                                                 |
|--------------------------|-------------|------------|----------------------------------------------------------------------|----------------------------------------------------------------------|
| Recovery codes           | None        | High       | - Zero external deps<br>- Standard in OSS (GitHub, Bitwarden)<br>- Fully offline | - Users lose them<br>- Must be shown once at setup<br>- No recovery if lost |
| TOTP / Authenticator     | None        | High       | - User already has the seed<br>- Works offline<br>- Widely understood | - Requires upfront setup<br>- Lost phone = lost access<br>- UX friction at onboarding |
| Passkeys / WebAuthn      | None        | Very high  | - No password at all<br>- Phishing-proof<br>- Device-native UX        | - Multi-device management is complex<br>- Browser/OS support varies<br>- New paradigm, may confuse users |
| Security questions       | None        | Low        | - Dead simple to implement<br>- No setup burden on user               | - Answers are guessable or public<br>- Discouraged by NIST<br>- Bad UX reputation |
| Admin / manual reset     | None        | High       | - Solid identity verification<br>- Good for self-hosted teams         | - Requires a human admin<br>- Doesn't scale<br>- Slow for the user |
| Trusted device approval  | Minimal     | High       | - Seamless if user has another session<br>- No codes to store         | - Useless if no active session exists<br>- Needs push or polling mechanism<br>- Complex to implement |
| Identity file / keyfile  | None        | Medium     | - Common in OSS / CLI tools (SSH, PGP)<br>- No server involved<br>- Power-user friendly | - File loss = account loss<br>- Bad UX for non-technical users<br>- No standard UI paradigm |

@Raphi-Sch any preferences ?

**Comment by Raphi-Sch** [comment]
TOTP / Auth and Recovery code for redundancy

**Comment by CharlesLgn** [comment]
# Password recovery workflows

## 1. Registration flow

```mermaid
flowchart LR
    A([User registers]) --> B[Generate recovery codes\n8–12 one-time codes, hashed in DB]
    B --> C[Show codes to user once\nWarn: save them now or lose access]
    C --> D{Set up TOTP?\noptional}
    D -->|skip| E([Account ready])
    D -->|yes| F[Verify TOTP\nConfirm 6-digit code]
    F --> G([Account ready + TOTP])
```

---

## 2. Recovery flow

```mermaid
flowchart LR
    A([User clicks forgot password]) --> B[Enter a recovery code\nOne of the 8–12 saved codes]
B --> C{Code valid?}
C -->|no| D([Deny access])
C -->|yes| E[Invalidate used code\nMark as consumed in DB]
E --> F{TOTP enabled\non account?}
F -->|no| G[Set new password]
F -->|yes| H[Verify TOTP]
H --> I{TOTP Valid?}
I -->|no| H
I -->|yes| G
G --> K([Access restored])
```

---

## 3. Recovery code regeneration flow

```mermaid
flowchart LR
    A([User opens security settings]) --> B[Clicks regenerate recovery codes]
    B --> C[Re-authenticate\nConfirm current password]
    C -->|wrong| D([Deny + warn])
    C -->|ok| E{TOTP enabled?}
    E -->|no| G
    E -->|yes| F[Verify TOTP\n6-digit code]
    F --> G[Invalidate all existing codes\nDelete or expire previous set in DB]
    G --> H([Generate + show new codes once])
```
